app/models/micropost.rb

```diff
@@ -0,0 +1,9 @@
+class Micropost < ApplicationRecord
+  belongs_to :user
+
+  scope :ordered_by_date, ->{order created_at: :desc}
+
+  validates :user_id, presence: true
```

Comment:
```suggestion
  validates :user, presence: true
```
sự khác biệt giữa user_id và user trong trường hợp này là gì nhỉ?

Reply:
micropost = Micropost.new
u = User.new
trường hợp :user_id micropost valid khi micropost.user_id có giá trị
còn :user thì micropost valid khi micropost.user = u. micropost valid kể cả khi u.id = null